imports/lib/schemas/typedSchemas.ts

```diff
@@ -241,14 +241,14 @@ export const inheritSchema = function <
   CT extends Record<string, any>,
   CP extends Record<keyof CT, t.Mixed>
 > (
-  parentSchemaType: t.InterfaceType<PP, PT>,
-  childSchemaType: t.InterfaceType<CP, CT>,
+  parentSchemaCodec: t.InterfaceType<PP, PT>,
+  childSchemaCodec: t.InterfaceType<CP, CT>,
   parentOverrides: Overrides<PT>,
   childOverrides: Overrides<CT>
-): [t.InterfaceType<PP & CP>, Overrides<PT & CP>] {
-  const inheritedType = t.type({
-    ...parentSchemaType.props,
-    ...childSchemaType.props,
+): [t.TypeC<PP & CP>, Overrides<PT & CP>] {
```

Comment:
This is the one actually meaningful change in this commit. `t.InterfaceType` captures the map of keys to codecs in its first type parameter, but the actual type is the second type parameter (which defaults to `any`, not very useful). `t.TypeC` is just a `t.InterfaceType` with better defaults: https://github.com/gcanti/io-ts/blob/6f75b1026921b87485332be5dabd3e51397af185/src/index.ts#L767-L768

Without this, taking `t.TypeOf<typeof AnnouncementCodec>` or whatever just returns `any`